### ISSUE TITLE
`reportDeprecated` not working in version 1.1.361

### ISSUE THREAD

**Issue by jzazo** [microsoft/pyright#7868]
**Describe the bug**
Pyright seems not to report `reportDeprecated: "error" as errors / warnings or set to true when set in config.

**Code or Screenshots**
```
from pathlib import Path

Path(__file__).link_to("example.py") 
```
When calling pyright from command line, it does not show an error. But `link_to` is deprecated and docs say:
`Deprecated since Python 3.10 and scheduled for removal in Python 3.12. Use hardlink_to() instead.`

**VS Code extension or command-line**
Both in vscode and cli.
Version: 1.1.361

**Comment by erictraut** [comment]
This method is not marked as deprecated in the typeshed stub [pathlib.pyi](https://github.com/python/typeshed/blob/3e3fd0dd86933214d903654c1e7057f25dba93fa/stdlib/pathlib.pyi#L225). If you think it should be, please file a bug (or submit a PR) in the [typeshed project](https://github.com/python/typeshed/issues).

**Comment by jzazo** [comment]
Thanks @erictraut for your response. I guess my concern comes on how to create deprecation warnings in code that actually warns users. Pycharm apparently reads the warnings inside the method's body and issue a warning. Should vscode/pylance do the same? Is this not pyrights's responsibility? I read somewhere that pylance was delegating this responsibility to pyright, that's why I created this issue here. Do you know who should have this responsibility?

**Comment by erictraut** [comment]
There is a (relatively recent) standard created for this. It's detailed in [PEP 702](https://peps.python.org/pep-0702/) and has been incorporated into the official [Python typing standard](https://typing.readthedocs.io/en/latest/spec/directives.html#deprecated). Pyright is a standards-based type checker and conforms to this specification.

The typeshed stubs already contain many `@deprecated` decorators, and pyright honors those. It appears that this one is just missing.
